integration-tests/load/ccip/destination_gun.go

```diff
@@ -63,9 +93,33 @@ func NewDestinationGun(
 	overrides *ccip.LoadConfig,
 	evmSourceKeys map[uint64]*bind.TransactOpts,
 	solanaSourceKeys map[uint64]*solana.PrivateKey,
-	chainOffset int,
 	metricPipe chan messageData,
+	laneConfig *crib.LaneConfiguration, // Lane configuration parameter
 ) (*DestinationGun, error) {
+	// Get available source chains based on lane configuration
+	var availableSources []uint64
+	if laneConfig != nil {
+		availableSources = laneConfig.GetSourceChainsForDestination(chainSelector)
+	}
+
+	// Fallback to any-to-any if no lane config or no sources found
+	if len(availableSources) == 0 {
```

Comment:
Hmm, I think it might be better to panic/error and enforce laneConfig is set. Implicitly defaulting to any-to-any might be tricky and hard to spot when param is missing